tests/test_caplugin.py

```diff
@@ -106,15 +105,20 @@ async def test_subscribe_disconnect(schema: Schema):
 
 @pytest.mark.asyncio
 async def test_subscribe_ticking(ioc: Popen, schema: Schema):
-    results = []
-    await caput(PV_PREFIX + "ticking", 0.0)
+    results: List[Optional[Dict[str, Any]]] = []
     resp = await schema.subscribe(ticking_subscription_query)
     assert isinstance(resp, AsyncIterator)
-    start = time.time()
+    startVal = 0.0
+    count = 0
     async for result in resp:
-        if time.time() - start > 1.0:
+        if count > 2:
             break
+        # Get the starting value in the subscription for checks later
+        if count == 0:
+            startVal = await caget(PV_PREFIX + "ticking")
```

Comment:
As above, is there a reason this cannot be brought outside the `for` loop?

Reply:
See above comment.